snooty/parser.py

```diff
@@ -1177,6 +1178,41 @@ def __init__(
 
         self.config.substitution_nodes = substitution_nodes
 
+        # Parse banner value and instantiate a banner node for postprocessing, if a banner value is defined.
+        if self.config.banner and self.config.banner.get("value"):
+            banner_data = self.config.banner
+
+            variant = (
+                banner_data.get("variant")
+                if isinstance(banner_data.get("variant"), str)
+                else "info"
+            )
+
+            target = (
+                ""
+                if not isinstance(banner_data.get("target"), str)
+                else banner_data.get("target")
+            )
+
+            assert target is not None and variant is not None
+            options = {"variant": variant}
+            self.config.banner_node = BannerConfig(
+                target, n.Directive((-1,), [], "", "banner", [], options)
```

Comment:
Is `-1` a sensible span value for an injected `Node`? In general, I'm unsure about if/where this value would be consumed after this step in the postprocessor.

Reply:
Mehhhh, I guess in theory we should actually have a tiny `n.Root` for things defined in external sources, but for now `-1` is an idiom we do use elsewhere.

If I change my mind about this, I can always change it :D